src/parser/jobs/ast/modules/Lightspeed.tsx

```diff
@@ -0,0 +1,234 @@
+import {t} from '@lingui/macro'
+import {Trans} from '@lingui/react'
+import {DataLink} from 'components/ui/DbLink'
+import {RotationTable} from 'components/ui/RotationTable'
+import {Status} from 'data/STATUSES'
+import {Event, Events} from 'event'
+import {Analyser} from 'parser/core/Analyser'
+import {EventHook} from 'parser/core/Dispatcher'
+import {filter} from 'parser/core/filter'
+import {dependency} from 'parser/core/Injectable'
+import {Data} from 'parser/core/modules/Data'
+import {Timeline} from 'parser/core/modules/Timeline'
+import React, {Fragment} from 'react'
+import {Icon, Message} from 'semantic-ui-react'
+import DISPLAY_ORDER from './DISPLAY_ORDER'
+
+const BASE_GCDS_PER_WINDOW = 6
+
+interface LIGHTSPEED_Window {
+	start: number
+	end?: number
+
+	rotation: Array<Events['action']>
+	gcdCount: number
+	mpSavings: number
+	trailingGcdEvent?: Events['action']
+
+	buffsRemoved: Array<Status['id']>
+	deathTruncated: boolean
+}
+
+// in this module we only want to track Lightspeed windows opened by
+// the character selected for analysis. windows that clip into
+// AST Lightspeed will be marked.
+// Used DNC Technicalities as basis for this module. Rewritten from previous module for consistency purposes
+export default class Lightspeed extends Analyser {
+	static override handle = 'Lightspeed'
+	static override title = t('ast.Lightspeed.title')`Lightspeed`
+	static override displayOrder = DISPLAY_ORDER.LIGHTSPEED
+
+	@dependency private timeline!: Timeline
+	@dependency private data!: Data
+
+	private history: LIGHTSPEED_Window[] = []
+	private currentWindow: LIGHTSPEED_Window | undefined = undefined
+	private castHook?: EventHook<Events['action']>
+
+	override initialise() {
+		const lightspeedFilter = filter<Event>().status(this.data.statuses.LIGHTSPEED.id)
+
+		this.addEventHook(lightspeedFilter.type('statusApply')
+			.target(this.parser.actor.id), this.tryOpenWindow)
+		this.addEventHook(lightspeedFilter.type('statusRemove')
+			.target(this.parser.actor.id), this.tryCloseWindow)
+
+		//this.addEventHook(filter<Event>().source(this.parser.actor.id).type('action'), this.onCast)
+	}
+
+	private tryOpenWindow(event: Events['statusApply']) {
+		if (this.currentWindow === undefined) {
+			this.currentWindow = {
+				start: event.timestamp,
+				rotation: [],
+				gcdCount: 0,
+				mpSavings: 0,
+
+				buffsRemoved: [],
+				deathTruncated: false,
+
+			}
+			this.castHook = this.addEventHook(
+				filter<Event>()
+					.source(this.parser.actor.id)
+					.type('action'),
+				this.onCast,
+			)
+		}
+
+		// If that happens, re-open the last window and keep tracking
+		if (this.currentWindow.end != null && this.currentWindow.end === event.timestamp) {
+			this.currentWindow.end = undefined
+		}
```

Comment:
What's the conditions where this can even happen?

Reply:
I think it was a situation that arose probably before?
I left it because while I wasn't sure of the situation doesn't mean it couldn't happen (whatever this situation was)
but anyway, I removed it